traits/trait_types.py

```diff
@@ -1901,58 +1899,96 @@ def init_fast_validate(self, *args):
+       case.
+
+    2. Instances of ``str``, ``bytes`` and ``bytearray`` are not treated
+       as collections for the purposes of this trait type, both for pragmatic
+       reasons (it's more likely that a user wants to use a string as an
+       element in a collection than as a collection in its own right), and
+       because the behavior of the ``in`` operator for those types does not
+       express the usual membership semantics (for example, ``"bc" in "abc"``
+       is ``True``).
 
     Parameters
     ----------
     *args
-        The enumeration of all legal values for the trait.  The expected
-        signatures are either:
-
-        - a collection.  The default value is the first item in the
-          collection. The collection should conform to the
-          collections.abc.Collection interface. That is, it at least
-          provides the __contains__, __len__ and __iter__ methods.
-          Note that although the types str, bytes, and bytearray
-          conform to the collection interface, these are handled
-          as discrete units.
-        - a single default value, combined with the values keyword
-          argument.
-        - a default value, followed by a single list enum.Enum, tuple or
-          collection conforming to collections.abc.Collection
-        - arbitrary positional arguments each giving a valid value.
-    values : str
-        The name of a trait holding the legal values.  A default value may
-        be provided via a positional argument, otherwise the first item in
-        the collection is used as the default value. Note that if the
-        collection does not have a notion of order like a set, the default
-        value will be an arbitrary element from the set.
+        The enumeration of all valid values for the trait. For a static
+        enumeration trait (where the *values* keyword argument is not given)
+        the supported signatures for ``args`` are as follows:
+
+        (collection,)
+            A nonempty collection of valid values. The default is the first
+            element of the collection, in iteration order.
+        (default, collection)
+            The default value, followed by a nonempty collection of valid
+            values. The default should be an element of the collection, but
+            this is not checked.
+        (item1, item2, ..., itemn)
+            One or more items giving the valid values for the collection.
+            The default is *item1*.
+
+        For a dynamic enumeration trait, where the *values* keyword argument
+        is given, the supported signatures for ``args`` are:
+
+        ()
+            No arguments given. In this case the default is the first item
+            of the collection, in iteration order.
+        (default,)
+            The default value for the collection.
+
+        For the static case, the ambiguity in the signatures is resolved
+        as follows: if ``args`` has length ``1`` or ``2``, ``arg[-1]`` can be
```

Comment:
Is it `args[-1]` in place of `arg[-1]` ?

Reply:
Thanks! Fixed.